daemon/api.go

```diff
@@ -1604,31 +1672,28 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 		return BadRequest("cannot create user: 'email' field is empty")
 	}
 
-	v, err := storeUserInfo(createData.Email)
-	if err != nil {
-		return BadRequest("cannot create user %q: %s", createData.Email, err)
+	// FIXME: check --known flag and use "getUserDetailsFromAssertions"
+	//        in this case and getUserDetailsFromStore otherwise
+	username, opts, err := getUserDetailsFromAssertion(c.d.overlord.State(), createData.Email)
+	if username == "" {
+		username, opts, err = getUserDetailsFromStore(createData.Email)
 	}
-	if len(v.SSHKeys) == 0 {
-		return BadRequest("cannot create user for %s: no ssh keys found", createData.Email)
+	if err != nil {
+		return BadRequest("%s", err)
 	}
+	opts.Sudoer = createData.Sudoer
+	opts.ExtraUsers = !release.OnClassic
 
-	gecos := fmt.Sprintf("%s,%s", createData.Email, v.OpenIDIdentifier)
-	opts := &osutil.AddUserOptions{
-		SSHKeys:    v.SSHKeys,
-		Gecos:      gecos,
-		Sudoer:     createData.Sudoer,
-		ExtraUsers: !release.OnClassic,
-	}
-	if err := osutilAddUser(v.Username, opts); err != nil {
-		return BadRequest("cannot create user %s: %s", v.Username, err)
+	if err := osutilAddUser(username, opts); err != nil {
+		return BadRequest("cannot create user %s: %s", username, err)
 	}
 
 	var createResponseData struct {
 		Username    string `json:"username"`
 		SSHKeyCount int    `json:"ssh-key-count"`
 	}
-	createResponseData.Username = v.Username
-	createResponseData.SSHKeyCount = len(v.SSHKeys)
+	createResponseData.Username = username
+	createResponseData.SSHKeyCount = len(opts.SSHKeys)
```

Comment:
I've reviewed this before, and it sounds like it didn't make its way into the original branch: rather than sending the count down, can we please send the actual keys? We'll need to sync with the console-conf folks for that, but it'd be nice to get that done this week still, before the freeze, otherwise we'll need to forever send keys+len(keys).